javascript/javascript1/week2/formal-fullname.js

```diff
@@ -0,0 +1,49 @@
+// In order to add this optional formal title before fullname, I write another function
+// i use "!" not logical operator to check if name & surname part is empty
+
+function getFullname(firstname, surname, useFormalName) {
+  if (!firstname && !surname) {
+    return "Name required!";
+  }
+
+  let fullname = firstname + " " + surname;
+
+  // here I write a conditional to add lord if useFormalName is true
+  if (useFormalName) {
+    fullname = "Lord" + fullname;
+  }
+  return fullname;
+}
+
+let fullname3 = getFullname(" Malcolm", "X", true);
```

Comment:
You can't rely on input always having empty space between the name. I suggest you do the implementation on the string above with "Lord " instead.
```suggestion
const fullname3 = getFullname("Malcolm", "X", true);
```
Second thing, all those fullname1 to fullname6 can be set as const, as you never reassign them.